src/main/java/org/alliancegenome/mati/entity/SubdomainEntity.java

```diff
@@ -0,0 +1,32 @@
+package org.alliancegenome.mati.entity;
+
+
+import lombok.Data;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.validation.constraints.NotEmpty;
+
+@Entity(name = "Subdomain")
+@Table(name = "subdomain")
+@Data
+public class SubdomainEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Integer id;
```

Comment:
If you create a BaseEntity class that all these extend you can move this id field in there and only have to define it once, and every class can make use of it. Also you might want to have id be `Long` vs `Integer` you will get much more id space that way.

Reply:
I am looking into this extension, doing something similar: https://quarkus.io/guides/rest-data-panache

Reply:
Done inheriting from  PanacheEntity